### PR TITLE
Fix: Add status in group member modal

### PR DESCRIPTION
**Pull request by Boruus**
#### Summary
- Add `StatusIcon` component in `GroupMemberList` and `ViewUserGroupModal` components.
- We get status by using `getStatusForUserId` selector.

#### Ticket Link
No ticket or issue are link to this change.

#### Screenshots
(I had a little trouble testing the fix in a local environment because the feature `group member` is only accessible from an enterprise offer, but it looks something like this: )
|  before  |  after  |
|----|----|
| image| image |

#### Release Note

```release-note
Fix: Add status in group member modal
```

### CODE REVIEW

**Comment by pvev** [comment]
hello @Boruus , please sign the [contributor license agreement](https://mattermost.com/mattermost-contributor-agreement/) so we can proceed with the review and merge of this PR.

**Comment by Boruus** [comment]
> hello @Boruus , please sign the [contributor license agreement](https://mattermost.com/mattermost-contributor-agreement/) so we can proceed with the review and merge of this PR.
Capture d’écran 2024-07-30 à 12 04 07

/check cla

File: webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx [hunk 1]
```diff
@@ -149,6 +155,8 @@ const GroupMemberList = (props: Props) => {
     };
 
     const Item = ({index, style}: ListChildComponentProps) => {
+        const status = useSelector((state: GlobalState) => getStatusForUserId(state, members[index].user.id) || UserStatuses.OFFLINE);
```
Comment (pvev) [inline webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:158]:
```suggestion
        const status = useSelector((state: GlobalState) => getStatusForUserId(state, members[index]?.user?.id) || UserStatuses.OFFLINE);
```

I pull your changes and it was breaking when clicking to see the group information. I identified this line as the problem, with this optional values the issue is solved 
Screenshot 2024-08-07 at 23 58 51

**Review by pvev — CHANGES_REQUESTED** [review]
thanks for your contribution @Boruus 🚀 , so far is a great work, there is only an issue happening. Please address my comments and let me know if you need any assistance. Thanks

**Comment by Boruus** [comment]
> thanks for your contribution @Boruus 🚀 , so far is a great work, there is only an issue happening. Please address my comments and let me know if you need any assistance. Thanks

Thanks for your feedback, I've added your changes which fix the problem nicely, I've also adjusted the button's css so that the overlay doesn't overlap the other users. Don't hesitate if you have any other feedback. 
Capture d’écran 2024-08-08 à 09 23 28

File: webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx [hunk 2]
```diff
@@ -171,13 +179,20 @@ const GroupMemberList = (props: Props) => {
...
+                            <span className='status-wrapper'>
+                                <Avatar
+                                    username={user.username}
+                                    size={'sm'}
+                                    url={Utils.imageURLForUser(user?.id ?? '')}
+                                    className={'avatar-post-preview'}
+                                    tabIndex={-1}
+                                />
+                                <StatusIcon
+                                    className='status user-popover-status'
```
Comment (larkox) [inline webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:191]: Maybe I am missing something, but the only place `user-popover-status` is defined, is in `webapp/channels/src/components/profile_popover/profile_popover.scss`, and it is defined under the `user-popover-image` class, so... it should not be applying any stile here, right? If so, why the class name?

File: webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx [hunk 3]
```diff
@@ -171,13 +179,20 @@ const GroupMemberList = (props: Props) => {
...
+                                    status={status}
+                                    button={true}
```
Comment (larkox) [inline webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx:193]: The `button` property seems to only remove the `status` class, and we are adding it again in the class name. Therefore... shouldn't it work the same way omitting the `button` prop and the `status` part of the classname?

File: webapp/channels/src/components/view_user_group_modal/view_user_group_list_item/view_user_group_list_item.tsx [hunk 4]
```diff
@@ -41,19 +47,27 @@ const ViewUserGroupListItem = (props: Props) => {
...
+
                 <Avatar
                     username={user.username}
                     size={'sm'}
                     url={Utils.imageURLForUser(user?.id ?? '')}
                     className={'avatar-post-preview'}
                 />
-            </>
+                <StatusIcon
+                    className='status bottom user-popover-status'
```
Comment (larkox) [inline webapp/channels/src/components/view_user_group_modal/view_user_group_list_item/view_user_group_list_item.tsx:66]: Same as my other two comments.

**Review by larkox — CHANGES_REQUESTED** [review]
Looks great, but a couple of things to verify.

Also, almost sure there are still some lint issues, so please run `npm run fix` .

File: webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx (hunk 2 above)
Comment (Boruus) [reply webapp/channels/src/components/user_group_popover/group_member_list/group_member_list.tsx]: @larkox you're right the props classname and button are useless in these cases, I thought I had to add a style for the status icon but there's already a `.status` included in `.status-wrapper` class so no need to add it, thanks for the feedback

**Comment by Boruus** [comment]
snapshot are failing i'm fixing it

**Comment by Boruus** [comment]
> snapshot are failing i'm fixing it

fixed

**Review by pvev — APPROVED** [review]
Thanks for addressing our comments. This looks great now 🎉, Approved ✅  . Thank you for this fantastic work and thank you for contributing to Mattermost 🚀

**Review by larkox — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
image

**Pull request merged**